### PR TITLE
handle crates when they are not specified for std docs

### LINKED ISSUE #128610: [rustdoc-json] Regression — `core.json`, `alloc.json` and `std.json` are no longer available on nightly for `rust-docs-json` component
I have pinned down the regression to a specific commit, thanks to @aDotInTheVoid recommending me `cargo-bisect-rustc`.

## Output of `cargo bisect-rustc`

searched nightlies: from nightly-2024-07-28 to nightly-2024-07-30
regressed nightly: nightly-2024-07-30
searched commit range: https://github.com/rust-lang/rust/compare/2cbbe8b8bb2be672b14cf741a2f0ec24a49f3f0b...612a33f20b9b2c27380edbc4b26a01433ed114bc
regressed commit: https://github.com/rust-lang/rust/commit/56c698c71130de6fe55ba703a161405b6145b90e

<details>
<summary>bisected with <a href='https://github.com/rust-lang/cargo-bisect-rustc'>cargo-bisect-rustc</a> v0.6.9</summary>

Host triple: aarch64-apple-darwin
Reproduce with:
```bash
cargo bisect-rustc -c rust-docs-json --start=2024-07-28 --end=2024-07-30 --script ./regr.sh
```

`regr.sh`:

```bash
#!/bin/sh
set -ex

if [ ! -f ~/.rustup/toolchains/${RUSTUP_TOOLCHAIN}/share/doc/rust/json/core.json ]; then
    echo "File not found!"
    exit 1
fi
```

</details>

### PR DESCRIPTION
**Pull request by onur-ozkan**
Fixes a regression from https://github.com/rust-lang/rust/pull/128182.

Resolves #128610

### CODE REVIEW

**Comment by aDotInTheVoid** [comment]
question (as someone who doesn’t understand bootstrap/rustup components): is it possible to add a test for this? can we ensure that the `rust-docs-json` is gaurenteed to have `std.json` and this doesn’t regress again?

**Comment by onur-ozkan** [comment]
> question (as someone who doesn’t understand bootstrap/rustup components): is it possible to add a test for this? can we ensure that the `rust-docs-json` is gaurenteed to have `std.json` and this doesn’t regress again?

There are a few ways to test it, but I wasn't sure where to handle/test it. The best way might be to mock the bootstrap tests, but that requires huge work. [This](https://github.com/rust-lang/rust/pull/128631/commits/b64260eb81930b4e90ff6c502d1b3c2ecf93a7a4) should be enough until then.

**Comment by aDotInTheVoid** [comment]
Actual failure:

```
  
  ---- core::builder::tests::dist::dist_baseline stdout ----
  thread 'core::builder::tests::dist::dist_baseline' panicked at src/core/build_steps/dist.rs:117:13:
  rust-docs-json is missing `alloc.json`.
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  
  ---- core::builder::tests::dist::dist_with_empty_host stdout ----
  thread 'core::builder::tests::dist::dist_with_empty_host' panicked at src/core/build_steps/dist.rs:117:13:
  rust-docs-json is missing `alloc.json`.
  
  ---- core::builder::tests::dist::dist_with_hosts stdout ----
  thread 'core::builder::tests::dist::dist_with_hosts' panicked at src/core/build_steps/dist.rs:117:13:
  rust-docs-json is missing `alloc.json`.
  
  ---- core::builder::tests::dist::dist_with_same_targets_and_hosts stdout ----
  thread 'core::builder::tests::dist::dist_with_same_targets_and_hosts' panicked at src/core/build_steps/dist.rs:117:13:
  rust-docs-json is missing `alloc.json`.
  
  ---- core::builder::tests::dist::dist_with_targets stdout ----
  thread 'core::builder::tests::dist::dist_with_targets' panicked at src/core/build_steps/dist.rs:117:13:
  rust-docs-json is missing `alloc.json`.
  
  ---- core::builder::tests::dist::dist_with_targets_and_hosts stdout ----
  thread 'core::builder::tests::dist::dist_with_targets_and_hosts' panicked at src/core/build_steps/dist.rs:117:13:
  rust-docs-json is missing `alloc.json`.
  ```

File: src/bootstrap/src/core/build_steps/dist.rs [hunk 1]
```diff
@@ -110,11 +110,18 @@ impl Step for JsonDocs {
         ));
 
         let dest = "share/doc/rust/json";
+        let out = builder.json_doc_out(host);
+
+        // Make sure these are present in the component.
+        #[cfg(not(feature = "bootstrap-self-test"))]
+        for f in ["alloc.json", "core.json", "std.json"] {
+            assert!(out.join(f).exists(), "rust-docs-json is missing `{f}`.");
```
Comment (onur-ozkan) [inline src/bootstrap/src/core/build_steps/dist.rs:118]: I guess we should do this only for the t1 targets?

**Comment by onur-ozkan** [comment]
Reverted 2nd commit to unblock this hotfix. We can add that sanity check later.

@bors r=Mark-Simulacrum rollup

**Pull request merged**